src/components/ui/Button/IconButton.tsx

```diff
@@ -3,12 +3,19 @@
 
 import React, { forwardRef } from 'react';
 import Button, { ButtonProps } from './';
+import { StyledIconButtonWrapper } from './Styled';
 
-export interface IconButtonProps extends ButtonProps {}
+export interface IconButtonProps extends ButtonProps {
+  /** Render a component to the top right area of the IconButton */
+  badge?: React.ReactNode | React.ReactNode[];
+}
 
 export const IconButton = forwardRef(
   (props: IconButtonProps, ref: React.Ref<HTMLButtonElement>) => (
-    <Button ref={ref} variant="icon" {...props} />
+    <StyledIconButtonWrapper>
+      <Button ref={ref} variant="icon" {...props} />
+      {props.badge && props.badge}
+    </StyledIconButtonWrapper>
```

Comment:
Do you foresee any isues with spreading props to the inner component and not the wrapper? Might get some weird behavior with something like <IconButton className="pos-absolute" />, since it'll be applied to the button but not the wrapper

Since it's pretty easy to work around, not a big dea but figured I'd bring it up

Reply:
It's a good point. In this case the wrapper element is purely for providing a new relative positioning context to allow for the button and the badge component to be positioned correctly. Ideally that never gets overridden.

One of the downsides to putting all of the props in the wrapper is that we've based a lot of our props on native HTML attributes, and if we were to do what you'd suggest, then `StyledIconButtonWrapper` would need to be able to accept `button` attributes, which i suspect will require a decent amount of TS wrangling, and ultimately result in us obfuscating which element is which.

Reply:
Yeah that makes sense to me, and might just make the component more confusing if you try to split the two props up. Sounds good to me